.gitlab-ci.yml

```diff
@@ -0,0 +1,113 @@
+    - echo "Boost version $(./boost_version)"
+    - export cupla_DIR=$CI_PROJECT_DIR
+    # use one build directory for all build configurations
+    - mkdir build
+    - cd build
+    # ALPAKA_ACCS contains the backends, which are used for each build
+    # the backends are set in the sepcialized base jobs .base_gcc,.base_clang and.base_cuda
+    - for CMAKE_FLAGS in $ALPAKA_ACCS ; do
+        echo "###################################################"
+        && echo "# Example Matrix Multiplication (adapted original)"
+        && echo "###################################################"
+        && echo "can not run with CPU_B_SEQ_T_SEQ due to missing elements layer in original SDK example"
+        && echo "CPU_B_SEQ_T_OMP2/THREADS too many threads necessary (256)"
+        && if [[ $CMAKE_FLAGS =~ -*DALPAKA_ACC_GPU_CUDA_ENABLE=ON.* ]]; then
+          cmake $cupla_DIR/example/CUDASamples/matrixMul/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+          && make -j
+          && ./matrixMul -wA=64 -wB=64 -hA=64 -hB=64
+          && rm -r * ;
+        fi
+        && echo "###################################################"
+        && echo "# Example Async API (adapted original)"
+        && echo "###################################################"
+        && echo "can not run with CPU_B_SEQ_T_SEQ due to missing elements layer in original SDK example"
+        && echo "CPU_B_SEQ_T_OMP2/THREADS too many threads necessary (512)"
+        && if [[ $CMAKE_FLAGS =~ -*DALPAKA_ACC_GPU_CUDA_ENABLE=ON.* ]]; then
+          cmake $cupla_DIR/example/CUDASamples/asyncAPI/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+          && make -j
+          && ./asyncAPI
+          && rm -r * ;
+        fi
+        && echo "###################################################"
+        && echo "# Example Async API (added elements layer)"
+        && echo "###################################################"
+        && cmake $cupla_DIR/example/CUDASamples/asyncAPI_tuned/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+        && make -j
+        && ./asyncAPI_tuned
+        && rm -r *
+        && echo "###################################################"
+        && echo "Example vectorAdd (added elements layer)"
+        && echo "###################################################"
+        && cmake $cupla_DIR/example/CUDASamples/vectorAdd/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+        && make -j
+        && ./vectorAdd 100000
+        && rm -r * ;
+      done
+
+.base_gcc:
+  variables:
+    GIT_SUBMODULE_STRATEGY: normal
+    CXX: g++
+    CC: gcc
+    ALPAKA_ACCS: "-DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE=ON
+                  -DALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE=ON
+                  -DALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE=ON"
+                  # -DALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE=ON
+  before_script:
+    - echo "number of processor threads $(nproc)"
+    - $CXX --version
```

Comment:
before_script is the same for every compiler at the moment, maybe it can be moved to general base job section

Reply:
This is not correct. The  base_clang has:
```bash
CXX: clang++                                                                                                       
CC: clang 
```

Reply:
I know, but I hoped this is handled by the variable $CXX. But it is probably not correctly set until you write the compiler specification.

Reply:
Sorry, I misunderstood your comment. Yes, I will move the `echo "numer` and the `$CXX --version` to script.

Reply:
fixed